Readme Update

thisiswhereitype opened the pull request:
Cleaned up throughout, added table of contents, updated support and refreshed examples.

thisiswhereitype committed:
Readme changes, cleaned up throughout, added contents, updated support and refreshed examples

snowleopard commented on README.md:
```diff
@@ -2,44 +2,31 @@
-
-All parsers currently export the following:
+Pangraph is a Haskell library which offers for parsing and serializations for  
+graph network files. As well as conversions to other Haskell graph formats.
```
graph network files -> graph files ('graph' and 'network' are mostly synonyms).

snowleopard commented on README.md:
```diff
@@ -2,44 +2,31 @@
+graph network files. As well as conversions to other Haskell graph formats.
+Some example formats are
+[GraphML](http://graphml.graphdrawing.org/)
+and [Workcraft](https://www.workcraft.org/).
```
Workcraft has a huge website, where exactly should I look for "example formats"?

benmvp replied:
Fixed: the Runtime now rejects observedState without observations, with a dedicated unit test.

snowleopard commented on README.md:
```diff
@@ -2,44 +2,31 @@
+3. [Library support](#graph-library-support)
+
+## Usage
+Pangraph offers an api in the module `Pangraph` for construction and   
```
api -> API

snowleopard commented on README.md:
```diff
@@ -2,44 +2,31 @@
+
+## Usage
+Pangraph offers an api in the module `Pangraph` for construction and   
+manipulation of graph networks. The parsers in the library use this module to   
```
graph networks -> graphs (do this across the README).

snowleopard commented:
Thanks, see a couple of minor comments above.

thisiswhereitype committed:
Clarifications and semantic changes to readme

thisiswhereitype commented:
I have addressed all your comments. I have removed the Workcraft link from the description as it does not have support right now and the directed graph section of the Workcraft website is a link to the Wikipedia page. Thanks for the feedback.

snowleopard commented:
@thisiswhereitype You use both 'graph' and 'network' terms which may potentially be confusing for the reader who may be wondering: Is there a difference between graphs and networks? If yes, what is it? etc. Since the library is called `pangraph` not `pannetwork` I suggest you stick to 'graphs' throughout the README. Imagine you're a newcomer who is reading the README -- try to make get rid of any potential sources of confusion. Maybe add a simple picture of what a graph is (for example, the one from our paper -- with four nodes A-E).

thisiswhereitype committed:
Corrections to spelling and standardisation of terms, image of example graph included

snowleopard commented on README.md:
```diff
@@ -1,10 +1,10 @@
-example format is [GraphML](http://graphml.graphdrawing.org/).
+Pangraph is a Haskell library which offers for parsing and serializations for graph files. As well as conversions to other Haskell graph formats. An example format is [GraphML](http://graphml.graphdrawing.org/). A graphml file
+could for example represent the following graph, with vertices from `A` to `E` and connections, the edges, between them.
+![a-sample-graph](examples/graphs/graph-AE.png?raw=true)
```
@thisiswhereitype Please use a vector (svg) image instead and scale it down -- it's too large at the moment!

snowleopard commented:
@thisiswhereitype I think you use this file directly: https://github.com/tuura/fdl17-paper/blob/master/fig/network.svg, but you'll need to copy it into this repository.

snowleopard commented on README.md:
```diff
@@ -1,45 +1,30 @@
+Pangraph is a Haskell library which offers for parsing and serializations for graph files. As well as conversions to other Haskell graph formats. An example format is [GraphML](http://graphml.graphdrawing.org/). A graphml file
+could for example represent the following graph, with vertices from `A` to `E` and connections, the edges, between them.
+![a-sample-graph](examples/graphs/graph-AE.png?raw=true)
+Source:
```
No need to acknowledge the source of the image in this case since I'm a co-author in both cases.

thisiswhereitype committed:
Changed the graph example to svg

thisiswhereitype commented:
I am happy for this to be merged if you also are.

snowleopard commented:
@thisiswhereitype Thanks, merged!
